### PR TITLE
finagle-{netty-http, http2, http}: -> 2.13

### PR DESCRIPTION
**Pull request by martijnhoekstra**
Problem

finagle-{netty-http, http2, http} no 2.13 build

Solution

cherry-pick commits of by @spocks from https://github.com/martijnhoekstra/finagle/pull/1

Result

finagle-{netty-http, http2, http}: -> 2.13

cc @spockz

### CODE REVIEW

File: finagle-http/src/test/scala/com/twitter/finagle/http/TracingTest.scala [hunk 1]
```diff
@@ -159,7 +159,7 @@ class TracingTest extends FunSuite with ScalaCheckDrivenPropertyChecks {
   }
 
   test("hasAllRequiredHeaders with all") {
-    forAll(Gen.someOf(Header.Required :+ "lol")) { headers: Seq[String] =>
+    forAll(Gen.someOf(Header.Required :+ "lol")) { headers: scala.collection.Seq[String] =>
```
Comment (mosesn) [inline finagle-http/src/test/scala/com/twitter/finagle/http/TracingTest.scala]: can we make this immutable?

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala [hunk 2]
```diff
@@ -87,11 +87,12 @@ class Http2ListenerTest extends FunSuite {
                   |
                   |""".stripMargin.replaceAll("\n", "\r\n")))
 
-    assert(await(read()).get == """HTTP/1.0 400 Bad Request
-                                  |Connection: close
-                                  |Content-Length: 0
-                                  |
-                                  |""".stripMargin.replaceAll("\n", "\r\n"))
+    val responseLines = await(read()).get.lines.toIndexedSeq // not sure whether lines plays nice with JDK11, would only break running the *tests* anyway?
```
Comment (mosesn) [inline finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala]: why did we have to change this?  can we revert this change?  also this comment is hard to parse, is it still supposed to be here?

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/transport/client/UpgradeRequestHandlerTest.scala [hunk 3]
```diff
@@ -56,7 +56,7 @@ class UpgradeRequestHandlerTest extends FunSuite {
       val request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/")
       channel.writeOutbound(request)
 
-      assert(stats.counters.get(Seq("upgrade", "attempt")) == Some(1l))
+      assert(stats.counters.get(Seq("upgrade", "attempt")).contains(1L))
```
Comment (mosesn) [inline finagle-http2/src/test/scala/com/twitter/finagle/http2/transport/client/UpgradeRequestHandlerTest.scala]: what was wrong with the original version?

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala (hunk 2 above)
Comment (yufangong) [reply finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala]: +1

**Review by yufangong — COMMENTED** [review]
LGTM after reverting some test changes.

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala (hunk 2 above)
Comment (spockz) [reply finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala]: This was changed because the order of the headers differs between 2.12 and 2.13. (Or at least differed when I was changing/fixing the tests.)

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/transport/client/UpgradeRequestHandlerTest.scala (hunk 3 above)
Comment (spockz) [reply finagle-http2/src/test/scala/com/twitter/finagle/http2/transport/client/UpgradeRequestHandlerTest.scala]: Not sure. Happy to revert.

File: finagle-http/src/test/scala/com/twitter/finagle/http/TracingTest.scala (hunk 1 above)
Comment (martijnhoekstra) [reply finagle-http/src/test/scala/com/twitter/finagle/http/TracingTest.scala]: reverted to plain `Seq`

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala (hunk 2 above)
Comment (martijnhoekstra) [reply finagle-http2/src/test/scala/com/twitter/finagle/http2/Http2ListenerTest.scala]: reverted test completed successfully in 2.12 and 2.13, so reverted

File: finagle-http2/src/test/scala/com/twitter/finagle/http2/transport/client/UpgradeRequestHandlerTest.scala (hunk 3 above)
Comment (martijnhoekstra) [reply finagle-http2/src/test/scala/com/twitter/finagle/http2/transport/client/UpgradeRequestHandlerTest.scala]: `1l` is depricated due to poor legibility, so `1l` -> `1L`, otherwise reverted. could be `Some(1)` too, `Some(1L) == Some(1)`, regardless of what you may think of that. The bikest of sheds.

**Review by mosesn — APPROVED** [review]
thanks!  looks great =)

**Comment by yufangong** [comment]
@mgodave has merged this at https://github.com/twitter/finagle/commit/0e8fcfc2b73411653e3122a1fbe6f94130cf7ff8. Thanks all!
